### ISSUE TITLE
rs_asio w/ Motu Pro Audio (MOTU 624) - Crashes on startup, connected via Thunderbolt

### ISSUE THREAD

**Issue by mossmoss** [mdias/rs_asio#237]
Hello, I'm having an issue with my Motu 624 via Thunderbolt on a Windows 10 PC and Rocksmith 2014 Remastered running via the Steam app. I'm trying to use a guitar.

Rocksmith starts when I have my interface tuned off. 

With interface on, it crashes before starting up, with or without rs_asio files in place. 

With rs_asio files in place and interface off, Rocksmith starts with warning about no audio output device detected.

I'm trying to start without any profile saved in Rocksmith. 

my ini file looks like this: 

[Config]
EnableWasapiOutputs=0
EnableWasapiInputs=0
EnableAsio=1

[Asio]
; available buffer size modes:
;    driver - respect buffer size setting set in the driver
;    host   - use a buffer size as close as possible as that requested by the host application
;    custom - use the buffer size specified in CustomBufferSize field
BufferSizeMode=driver
CustomBufferSize=

[Asio.Output]
Driver=MOTU Pro Audio
BaseChannel=0
AltBaseChannel=
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.0]
Driver=MOTU Pro Audio
Channel=1
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.1]
Driver=
Channel=1
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

[Asio.Input.Mic]
Driver=
Channel=1
EnableSoftwareEndpointVolumeControl=1
EnableSoftwareMasterVolumeControl=1
SoftwareMasterVolumePercent=100

log file: 

0.000 [INFO]   - Wrapper DLL loaded (v0.5.7)
0.000 [INFO]  PatchOriginalCode
0.397 [INFO]  Patching CoCreateInstance
0.397 [INFO]  Patch_CallAbsoluteAddress - num locations: 5
0.397 [INFO]  Patching call at 01C5AA85
0.397 [INFO]  Patching call at 01F03B6E
0.397 [INFO]  Patching call at 01F5C681
0.397 [INFO]  Patching call at 01FAEF75
0.397 [INFO]  Patching call at 01FAF40D
0.397 [INFO]  Patching PortAudio MarshalStreamComPointers
0.397 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.397 [INFO]  Patching call at 01F03059
0.397 [INFO]  Patching PortAudio UnmarshalStreamComPointers
0.398 [INFO]  Patch_CallRelativeAddress - num locations: 1
0.398 [INFO]  Patching call at 01F0310E
0.400 [INFO]  Patched_CoCreateInstance called: IID_IMMDeviceEnumerator
0.401 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eRender - role: eMultimedia
0.401 [INFO]  RSAggregatorDeviceEnum::UpdateAvailableDevices
0.401 [INFO]  AsioHelpers::FindDrivers
0.401 [INFO]    ASIO4ALL v2
0.401 [INFO]    FL Studio ASIO
0.401 [INFO]    Maschine MK3
0.402 [INFO]    MOTU M Series
0.402 [INFO]    MOTU Pro Audio
0.402 [INFO]    Yamaha Steinberg USB ASIO
0.402 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - output requesting ASIO driver: MOTU Pro Audio
0.402 [INFO]  Creating AsioSharedHost - dll: C:\Program Files (x86)\MOTU\Pro Audio\x86\MOTUProAudioASIO.dll
0.412 [INFO]    ASIO input channels info:
0.412 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Mic 1
0.412 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Guitar 1
0.412 [INFO]      2 - active: 0, channel: 2, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Analog 1
0.412 [INFO]      3 - active: 0, channel: 3, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Analog 2
0.412 [INFO]      4 - active: 0, channel: 4, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Loopback L
0.412 [INFO]      5 - active: 0, channel: 5, group: 0, isInput: 1, type: ASIOSTInt32LSB, name: Loopback R
0.412 [INFO]    ASIO output channels info:
0.412 [INFO]      0 - active: 0, channel: 0, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Multiple Outputs (TB) 1
0.412 [INFO]      1 - active: 0, channel: 1, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Multiple Outputs (TB) 2
0.412 [INFO]      2 - active: 0, channel: 2, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Out 3
0.412 [INFO]      3 - active: 0, channel: 3, group: 0, isInput: 0, type: ASIOSTInt32LSB, name: Out 4
0.412 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.412 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - input[0] requesting ASIO driver: MOTU Pro Audio
0.412 [INFO]  RSAsioDeviceEnum::UpdateAvailableDevices - OK
0.412 [INFO]  DebugDeviceEnum::UpdateAvailableDevices - 1 render devices, 1 capture devices
0.412 [INFO]    hr: S_OK
0.412 [INFO]    *ppEndpoint: 11AFD620
0.412 [INFO]  DebugDeviceEnum::GetDefaultAudioEndpoint - dataFlow: eCapture - role: eMultimedia
0.412 [INFO]    hr: S_OK
0.412 [INFO]    *ppEndpoint: 11AFD7D0
0.412 [INFO]  DebugDeviceEnum::EnumAudioEndpoints - dataFlow: eAll - dwStateMask: 1
0.412 [INFO]    hr: S_OK
0.412 [INFO]    *ppDevices: 139714E8
0.412 [INFO]  {ASIO Out} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.412 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetCount
0.412 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetAt - iProp: 0
0.412 [INFO]  {ASIO Out} DebugWrapperDevicePropertyStore::GetAt - iProp: 1
0.412 [INFO]  {ASIO Out} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.412 [INFO]  {ASIO Out} DebugWrapperEndpoint::GetDataFlow
0.412 [INFO]    *pDataFlow: eRender
0.412 [INFO]  {ASIO Out} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.412 [INFO]  {ASIO Out} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevice::OpenPropertyStore - stgmAccess: 0
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetCount
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 0
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 1
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 2
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 3
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevicePropertyStore::GetAt - iProp: 4
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevice::QueryInterface - riid: IID_IMMEndpoint
0.412 [INFO]  {ASIO IN 0} DebugWrapperEndpoint::GetDataFlow
0.412 [INFO]    *pDataFlow: eCapture
0.412 [INFO]  {ASIO IN 0} DebugWrapperDevice::Activate - Activate iid: IID_IAudioClient dwClsCtx: 1
0.412 [INFO]  {ASIO IN 0} DebugWrapperAudioClient<struct IAudioClient3>::GetDevicePeriod

mini dump: 

Microsoft (R) Windows Debugger Version 10.0.22473.1005 X86
Copyright (c) Microsoft Corporation. All rights reserved.

Loading Dump File [C:\Program Files (x86)\Steam\steamapps\common\Rocksmith2014\rocksmith2014_369567_crash_2021_12_17T23_4_50C0.mdmp]
Comment: '    MXL4,0-E0x0,0x0,1PSHDX4 '
User Mini Dump File: Only registers, stack and portions of memory are available

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*
Symbol search path is: srv*
Executable search path is: 
Windows 10 Version 19044 MP (12 procs) Free x86 compatible
Product: WinNt, suite: SingleUserTS
Edition build lab: 19041.1.amd64fre.vb_release.191206-1406
Machine Name:
Debug session time: Fri Dec 17 18:04:51.000 2021 (UTC - 5:00)
System Uptime: not available
Process Uptime: 0 days 0:00:06.000
................................................................
....................................
This dump file has an exception of interest stored in it.
The stored exception information can be accessed via .ecxr.
(4e80.4810): Access violation - code c0000005 (first/second chance not available)
For analysis of this file, run !analyze -v
eax=00000000 ebx=1cc61ca8 ecx=00000000 edx=00000001 esi=1cc61c58 edi=1cc61c68
eip=778138fc esp=0019cf80 ebp=0019cf8c iopl=0         nv up ei pl nz ac po nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00200212
ntdll!NtGetContextThread+0xc:
778138fc c20800          ret     8
0:000> !analyze -v
*******************************************************************************
*                                                                             *
*                        Exception Analysis                                   *
*                                                                             *
*******************************************************************************

KEY_VALUES_STRING: 1

    Key  : AV.Fault
    Value: Write

    Key  : Analysis.CPU.mSec
    Value: 8811

    Key  : Analysis.DebugAnalysisManager
    Value: Create

    Key  : Analysis.Elapsed.mSec
    Value: 11996

    Key  : Analysis.Init.CPU.mSec
    Value: 250

    Key  : Analysis.Init.Elapsed.mSec
    Value: 4273

    Key  : Analysis.Memory.CommitPeak.Mb
    Value: 228

    Key  : Timeline.Process.Start.DeltaSec
    Value: 6

    Key  : WER.OS.Branch
    Value: vb_release

    Key  : WER.OS.Timestamp
    Value: 2019-12-06T14:06:00Z

    Key  : WER.OS.Version
    Value: 10.0.19041.1

FILE_IN_CAB:  rocksmith2014_369567_crash_2021_12_17T23_4_50C0.mdmp

COMMENT:      MXL4,0-E0x0,0x0,1PSHDX4 

CONTEXT:  (.ecxr)
eax=00000000 ebx=13b343f0 ecx=00000000 edx=00000001 esi=db2659b0 edi=13bd4450
eip=01f0c281 esp=0019f9c0 ebp=0019fc0c iopl=0         nv up ei ng nz ac pe cy
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00210297
Rocksmith2014!Scaleform::System::~System+0x21f0a0:
01f0c281 899e44010000    mov     dword ptr [esi+144h],ebx ds:002b:db265af4=????????
Resetting default scope

EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 01f0c281 (Rocksmith2014!Scaleform::System::~System+0x0021f0a0)
   ExceptionCode: c0000005 (Access violation)
  ExceptionFlags: 00000000
NumberParameters: 2
   Parameter[0]: 00000001
   Parameter[1]: db265af4
Attempt to write to address db265af4

PROCESS_NAME:  Rocksmith2014.exe

WRITE_ADDRESS:  db265af4 

ERROR_CODE: (NTSTATUS) 0xc0000005 - The instruction at 0x%p referenced memory at 0x%p. The memory could not be %s.

EXCEPTION_CODE_STR:  c0000005

EXCEPTION_PARAMETER1:  00000001

EXCEPTION_PARAMETER2:  db265af4

STACK_TEXT:  
WARNING: Stack unwind information not available. Following frames may be wrong.
0019fc0c 01f0c6ae     00000001 0019fc28 00000020 Rocksmith2014!Scaleform::System::~System+0x21f0a0
0019fc30 01efdfef     118c215c 00000001 0aead510 Rocksmith2014!Scaleform::System::~System+0x21f4cd
0019fc50 01efe13f     01f92cea 0aead510 1bc4ece0 Rocksmith2014!Scaleform::System::~System+0x210e0e
0019fc6c 01f93022     00000000 0aead510 00000000 Rocksmith2014!Scaleform::System::~System+0x210f5e
0019fc84 01f5c85f     0134e7e0 00000000 00000001 Rocksmith2014!CAkRegistryMgr::GetObjAndAddref+0x254b2
0019fccc 01ca059a     01384938 01384968 01ca75dc Rocksmith2014!AK::IsRestoreSinkRequested+0x88c
0019fd40 0190dc10     0ae5b950 0aed5b00 40000000 Rocksmith2014!luaopen_epgmGFx+0x2a5ce
0019fd60 0190da80     0ae5b950 0ae5b9b0 0ae48da0 Rocksmith2014!luaopen_VenueBindings+0x49f08
0019fd78 0190d535     77200f00 0ae5b950 00000000 Rocksmith2014!luaopen_VenueBindings+0x49d78
0019fdd4 0190d429     0ae54410 0019fe00 0177ae1b Rocksmith2014!luaopen_VenueBindings+0x4982d
0019fed0 01ffeae7     00400000 00000000 05126928 Rocksmith2014!luaopen_VenueBindings+0x49721
0019ff64 0218701e     76be0706 0019ff80 771ffa29 Rocksmith2014!iZTrashMultibandDistortion::iZTrashMultibandDistortionEngine::CreateWithDefaultAllocator+0x1c698
0019ff70 771ffa29     0036d000 771ffa10 0019ffdc Rocksmith2014!AK::MemoryMgr::GetPoolName+0x1805c7
0019ff80 77807a9e     0036d000 52f018ac 00000000 kernel32!BaseThreadInitThunk+0x19
0019ffdc 77807a6e     ffffffff 77828a53 00000000 ntdll!__RtlUserThreadStart+0x2f
0019ffec 00000000     02187000 0036d000 00000000 ntdll!_RtlUserThreadStart+0x1b

SYMBOL_NAME:  Rocksmith2014!Scaleform::System::~System+21f0a0

MODULE_NAME: Rocksmith2014

IMAGE_NAME:  Rocksmith2014.exe

STACK_COMMAND:  ~0s ; .ecxr ; kb

FAILURE_BUCKET_ID:  INVALID_POINTER_WRITE_c0000005_Rocksmith2014.exe!Scaleform::System::_System

OS_VERSION:  10.0.19041.1

BUILDLAB_STR:  vb_release

OSPLATFORM_TYPE:  x86

OSNAME:  Windows 10

FAILURE_ID_HASH:  {2f5b061b-cf78-2a9f-bd8d-85c8b4782ca2}

Followup:     MachineOwner
---------

**Comment by mdias** [comment]
Hello,

It doesn't look like I will be of much help, but some quick googling shows you're not the first one hitting this crash.

- [link 1](https://forums.ubisoft.com/showthread.php/2295413-Game-isn-t-starting-on-Steam)
- [link 2](https://forums.ubisoft.com/showthread.php/2303790-Windows-10-latest-update-white-screen-crash)

The way it's described seems to be caused by something else installed on the machine and/or possibly a windows update, so other than a very annoying clean windows install there's not much I could suggest, sorry.
